pusher/pusher.go

```diff
@@ -100,12 +100,6 @@ func (p *Pusher) routeMessages(ctx context.Context, msgChan *chan interfaces.Kaf
 			} else {
 				l.Error("Game not found")
 			}
-		case <-ctx.Done():
-			l.Info("Context done. Will stop routing messages.")
-			return
-		case <-p.stopChannel:
-			l.Info("Stop channel closed. Will stop routing messages.")
-			return
```

Comment:
why are we removing this? don't we want to still stop on ctx.Done and stopChannel?

Reply:
The logic is we need to keep routing the buffered messages during the graceful shutdown. If we return here on `context.Done` or when the `stopChannel` is closed, the messages in the buffer would never be processed and the `WaitGroup` we have to shutdown gracefully would be pointless.